### PR TITLE
languages/julia: init

### PR DESCRIPTION
**Pull request by diniamo**
## Sanity Checking

[editorconfig]: https://editorconfig.org
[changelog]: https://github.com/NotAShelf/nvf/tree/main/docs/release-notes

- [x] I have updated the [changelog] as per my changes.
- [x] I have tested, and self-reviewed my code.
- Style and consistency
  - [x] I ran **Alejandra** to format my code (`nix fmt`).
  - [x] My code conforms to the [editorconfig] configuration of the project.
  - [x] My changes are consistent with the rest of the codebase.
- If new changes are particularly complex:
  - [x] My code includes comments in particularly complex areas
  - [ ] I have added a section in the manual.
  - [ ] _(For breaking changes)_ I have included a migration guide.
- Package(s) built:
  - [ ] `.#nix` (default package)
  - [ ] `.#maximal`
  - [ ] `.#docs-html`
- Tested on platform(s)
  - [x] `x86_64-linux`
  - [ ] `aarch64-linux`
  - [ ] `x86_64-darwin`
  - [ ] `aarch64-darwin`

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: docs/release-notes/rl-0.7.md [hunk 1]
```diff
@@ -192,6 +192,10 @@ To migrate to `nixfmt`, simply change `vim.languages.nix.format.type` to
 - Add C# support under `vim.languages.csharp`, with support for both
   omnisharp-roslyn and csharp-language-server.
 
+- Add Julia support under `vim.languages.julia`. Note that the entirety of
+  Julia is bundled with nvf, if you enabled the module, since there is no way
+  to provide only the LSP server.
```
Comment (horriblename) [inline docs/release-notes/rl-0.7.md:197]: This should probably be in the `vim.language.julia.lsp.enable` option description

**Review by horriblename — COMMENTED** [review]
LGTM aside from the tiny nitpick

File: modules/plugins/languages/julia.nix [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+          description = ''
+            Whether to enable Julia LSP support.
+
+            ::: {.note}
+            The entirety of Julia is bundled with nvf, if you enable this
+            option, since there is no way to provide only the LSP server.
+
+            If you want to avoid that, you have to change
+            [](#opt-vim.languages.julia.lsp.package) to use the Julia binary
+            in PATH (set it to `null`), and add the `LanguageServer` package to
```
Comment (NotAShelf) [inline modules/plugins/languages/julia.nix:89]:
```suggestion
            in {env}`PATH` (set it to `null`), and add the `LanguageServer` package to
```

File: modules/plugins/languages/julia.nix [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+        };
+        server = mkOption {
+          description = "Julia LSP server to use";
+          type = enum (attrNames servers);
+          default = defaultServer;
+        };
+
+        package = mkOption {
+          description = ''
+            Julia LSP server package, null to use the Julia binary in PATH, or
```
Comment (NotAShelf) [inline modules/plugins/languages/julia.nix:104]:
```suggestion
            Julia LSP server package, `null` to use the Julia binary in {env}`PATH`, or
```

For consistency: `null` is inline code in previous desc, but not here.

**Review by NotAShelf — CHANGES_REQUESTED** [review]
Two quick changes for consistency.

Not a deal-breaker, but I'd appreciate it if the order of `mkOption` args were 

```nix
{
  type = ...
  default = ...
  description = ...
}

**Review by NotAShelf — APPROVED** [review]

**Pull request merged**